util/util.go

```diff
@@ -163,13 +162,13 @@ func AddImageNames(store storage.Store, image *storage.Image, addNames []string)
 func GetFailureCause(err, defaultError error) error {
 	switch nErr := errors.Cause(err).(type) {
 	case errcode.Errors:
-		return cli.NewMultiError([]error(nErr)...)
+		if strings.Contains(nErr.Error(), "not authorized") {
+			return fmt.Errorf("unauthorized: authentication required")
+		}
+		return err
 	case errcode.Error, *url.Error:
 		return nErr
 	default:
-		// HACK: In case the error contains "not authorized" like in
-		//       https://github.com/containers/image/blob/master/docker/docker_image_dest.go#L193-L205
-		// TODO(bshuster): change "containers/images" to return "errcode" rather than "error".
 		if strings.Contains(nErr.Error(), "not authorized") {
```

Comment:
Why remove _only_ the comment? The string matching remains a hack, and it seems to me that the TODO item also remains valuable.

Also, per the other comment, after https://github.com/containers/image/pull/410 this probably does not work any more and needs updating.

Reply:
I've actually removed it completely, I don't think it's necessary anymore, but then again I haven't done my one last test yet....